test/python/algorithms/test_phase_estimator.py

```diff
@@ -0,0 +1,259 @@
+        hamiltonian = ((0.5 * X) + (0.6 * Y) + (0.7 * I)).to_pauli_op()
+        state_preparation = None
+        result = self.hamiltonian_pe(hamiltonian, state_preparation, evolution=MatrixEvolution())
+        phase_dict = result.filter_phases(0.2, as_float=True)
+        phases = list(phase_dict.keys())
+        self.assertAlmostEqual(phases[0], 1.490, delta=0.001)
+        self.assertAlmostEqual(phases[1], -0.090, delta=0.001)
+
+    def _setup_from_bound(self, evolution):
+        hamiltonian = self.hamiltonian_1
+        state_preparation = None
+        bound = 1.2 * sum([abs(hamiltonian.coeff * pauli.coeff) for pauli in hamiltonian])
+        backend = qiskit.BasicAer.get_backend('statevector_simulator')
+        qi = qiskit.utils.QuantumInstance(backend=backend, shots=10000)
+        phase_est = HamiltonianPhaseEstimation(num_evaluation_qubits=6,
+                                               hamiltonian=hamiltonian,
+                                               bound=bound,
+                                               quantum_instance=qi,
+                                               state_preparation=state_preparation,
+                                               evolution=evolution)
+        result = phase_est.estimate()
+        return result
+
+    def test_from_bound(self):
+        """HamiltonianPhaseEstimation with bound"""
+        result = self._setup_from_bound(MatrixEvolution())
+        phases = result.filter_phases()
+        with self.subTest('test phases has the correct length'):
+            self.assertEqual(len(phases), 2)
+        with self.subTest('test scaled phases are correct'):
+            self.assertEqual(list(phases.keys()), [1.5, -1.5])
+        phases = result.filter_phases(scaled=False)
+        with self.subTest('test unscaled phases are correct'):
+            self.assertEqual(list(phases.keys()), [0.25, 0.75])
+        with self.subTest('test most_likely_phase method'):
+            self.assertEqual(result.most_likely_eigenvalue, 1.5)
+            self.assertEqual(result.most_likely_phase, 0.25)
+
+    def test_trotter_from_bound(self):
+        """HamiltonianPhaseEstimation with bound and Trotterization"""
+        result = self._setup_from_bound(PauliTrotterEvolution(trotter_mode='suzuki', reps=3))
+        phase_dict = result.filter_phases(0.1)
+        phases = list(phase_dict.keys())
+        with self.subTest('test phases has the correct length'):
+            self.assertEqual(len(phases), 2)
+        with self.subTest('test phases has correct values'):
+            self.assertAlmostEqual(phases[0], 1.5, delta=0.001)
+            self.assertAlmostEqual(phases[1], -1.5, delta=0.001)
+
+
+class TestPhaseEstimation(QiskitAlgorithmsTestCase):
+    """Evolution tests."""
+
+    # pylint: disable=invalid-name
+    def one_phase(self, unitary_circuit, state_preparation=None, n_eval_qubits=6,
+                  backend=qiskit.BasicAer.get_backend('qasm_simulator')):
```

Comment:
I think default args should not be function calls since they get executed at import time, could you make this None instead and set it in the function?

Reply:
Done.